### ISSUE TITLE
export order - not fetching data as expected

### ISSUE THREAD

**Issue by JayneNew** [mattbanks/woocommerce-local-pickup-time#80]
Hello, I'm exporting orders with Advanced Order Export For WooCommerce by AlgolPlus.  The is _local_pickup_time_select data field is not fetching data as expected for export.  

For example a customer Pickup Time: April 17, 2020 6:30 pm, exports as 1587112200, and a Pickup Time: April 18, 2020 7:30 pm, exports as 1587202200. 

Should I be using a different order export plugin, or exporting that field in a different way? 

Thanks. 
Jayne 

ps: thanks for a great plugin.

**Comment by timnolte** [comment]
@JayneNew the value is stored as a timestamp. You'll have to convert this value to the appropriate date/time format according to your locale. You would need to do something along these lines:

```
 $pickup_plugin = Local_Pickup_Time::get_instance();
$pickup_time_formatted = $pickup_plugin->pickup_time_select_translatable($pick_uptime_value)
```
Where $pickup_time_value is that number you are seeing.

I'm not familiar with the plugin you are using, however, I have used WP All Export/Import and it has the ability on export to use PHP code transform the data on export. This would be the type of functionality you'd need to have with the plugin you are using

**Comment by JayneNew** [comment]
Thank you Tim, understood and appreciate the explanation. 

The exporter I'm using has a 'Custom PHP code to modify output' field.  

I've tried the code but get an error.  I did replace ($pick_uptime_value) with ($pickup_time_value).  Is there something else in there that needs correcting for this code to work? Thanks. 

$pickup_plugin = Local_Pickup_Time::get_instance();
$pickup_time_formatted = $pickup_plugin->pickup_time_select_translatable($pickup_time_value)

**Comment by timnolte** [comment]
@JayneNew can you give me the error you received? Also, did you assign the value from the `_local_pickup_time_select` field to `$pickup_time_value`?

**Comment by timnolte** [comment]
@JayneNew if you could provide some screenshots of what that part of the plugin looks like that might help too.

**Comment by JayneNew** [comment]
@timnolte Please see attached, no I didn't assign the value, not sure how or where to do so.  Thanks. 
<img width="1660" alt="Screen Shot 2020-04-13 at 1 58 00 pm" src="https://user-images.githubusercontent.com/63540282/79091099-59496900-7d8f-11ea-970c-034033c84b6c.png">
<img width="418" alt="Screen Shot 2020-04-13 at 1 58 08 pm" src="https://user-images.githubusercontent.com/63540282/79091108-5ea6b380-7d8f-11ea-8f5d-d13a0c556b83.png">
<img width="444" alt="Screen Shot 2020-04-13 at 1 58 58 pm" src="https://user-images.githubusercontent.com/63540282/79091110-5f3f4a00-7d8f-11ea-9f2b-5d3a7ea17731.png">
<img width="1533" alt="Screen Shot 2020-04-13 at 2 00 20 pm" src="https://user-images.githubusercontent.com/63540282/79091111-60707700-7d8f-11ea-99c0-2239162c3f70.png">

**Comment by JayneNew** [comment]
@timnolte did you see the above screenshots? 

Thanks

**Comment by timnolte** [comment]
@JayneNew yes, the first thing is that the 2nd line in the custom PHP box is missing an ending semicolon. However, beyond that there is going to be more code needed that falls in line with the requirement of how that plugin uses custom PHP code. The documentation for that functionality would be found here: https://docs.algolplus.com/algol_order_export/overview-custom-code/ and more specifically using this sort of example: https://docs.algolplus.com/algol_order_export/fields/

**Comment by timnolte** [comment]
@JayneNew were you able to get some code working for your export?

**Comment by JayneNew** [comment]
Hello @timnolte thanks for following up.  No I wasn't, I tried but I'm a complete Newb so didn't want to be too annoying.  I'm not locked into that export plugin, so if there is one that you know will work with simple coding I'm happy to change.  

Also, I would like to discuss some small developments to your plugin, happy to discuss payment for further development - We, like many hospitality businesses, are changing with the current Covid times and offering take-away only.  The changes I have in mind would help many in the hospitality industry and beyond to schedule local pick-ups.  Let me know if this is of interest to you.

**Comment by timnolte** [comment]
@JayneNew are you using the [Lite](https://wordpress.org/plugins/woo-order-export-lite/) or [Pro](https://algolplus.com/plugins/downloads/advanced-order-export-for-woocommerce-pro/) version of the export plugin? If you are only using the Lite version that I can probably download it and test this out locally and provide some code that should work for you.

As for the additional enhancements that you'd like to see I'd encourage you to open up additional issues for those and I can look at prioritizing them. While I'm honored that you are willing to discuss some sort of compensation for development time I'm not sure I'm ready to step into that given this is a free plugin that I co-maintain with the original developer and it's hard for me to commit to hard timelines atm. As of course you are seeing yourself this plugin has picked up a lot of traction recently given the impact COVID-19 has had on businesses. I am starting on a 1.4 release of the plugin and my focus is going to be those items that are impacting a lot of business related to operating under the constraints of COVID-19. Thanks!

**Comment by JayneNew** [comment]
Hello @timnolte, thanks for your msg.  I'm just using the Lite version. 

I'll open up new issue for my suggestions for the Plugin, thanks!
